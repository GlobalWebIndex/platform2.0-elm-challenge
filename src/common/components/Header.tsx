import React from 'react';
import {
  Box,
  Flex,
  HStack,
  Button,
  Link,
  useDisclosure,
  Stack,
} from '@chakra-ui/react';
import { Link as ReactRouterLink } from 'react-router-dom';

interface Props {
    name: string,
    url: string
}
export const NavLink = ({ name, url }: Props): JSX.Element => (
  <Link
    as={ReactRouterLink}
    px={2}
    py={1}
    rounded="md"
    color="white"
    _hover={{
      textDecoration: 'none',
      bg: 'blue.900',
    }}
    to={url}
  >
    {name}
  </Link>
);

export const Header = (): JSX.Element => {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <Box bg="blue.500" px={4}>
      <Flex h={16} alignItems="center" justifyContent="space-between">
        <Button
          size="md"
          aria-label="Open Menu"
          display={{ md: 'none' }}
          onClick={isOpen ? onClose : onOpen}
        >
          {isOpen ? 'Close' : 'Menu'}
        </Button>
        <HStack spacing={8} alignItems="center">
          <Box color="white" fontWeight={500} mx={5}>CATS APP</Box>
          <HStack
            as="nav"
            spacing={4}
            display={{ base: 'none', md: 'flex' }}
          >
            <NavLink name="Images" url="/" />
            <NavLink name="Breeds" url="/breeds" />
            <NavLink name="Favourites" url="/favourites" />
          </HStack>
        </HStack>
      </Flex>

      {isOpen ? (
        <Box pb={4} display={{ md: 'none' }}>
          <Stack as="nav" spacing={4}>
            <NavLink name="Images" url="/" />
            <NavLink name="Breeds" url="/breeds" />
            <NavLink name="Favourites" url="/favourites" />
          </Stack>
        </Box>
      ) : null}
    </Box>
  );
};
