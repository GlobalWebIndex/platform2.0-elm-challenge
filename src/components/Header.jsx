import { makeStyles } from "@material-ui/core/styles";

import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";

import { Link } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  logo: {
    height: 48,
    borderRadius: "12px 0 12px 0",
    padding: 4
  },
  title: {
    flexGrow: 1
  }
}));

export default function Header() {
  const classes = useStyles();
  return (
    <AppBar position="static" className={classes.appBar}>
      <Toolbar>
        <Link to="/">
          <img
            alt="Plato's cat"
            src="https://i.imgur.com/XFV8I3d.png"
            className={classes.logo}
          />
        </Link>
      </Toolbar>
    </AppBar>
  );
}
