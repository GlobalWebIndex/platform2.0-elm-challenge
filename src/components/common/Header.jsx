import React from "react";
import PropTypes from "prop-types";

const Header = ({ title }) => {
  return !title ? null : (
    <header data-testid="header" className="title">
      <h1>{title}</h1>
    </header>
  );
};

Header.propTypes = {
  title: PropTypes.string,
};

export default Header;
