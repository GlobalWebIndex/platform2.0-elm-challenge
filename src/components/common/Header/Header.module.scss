ul {
  list-style: none;
  display: flex;
  justify-content: space-around;
  background-color: lightgray;
  border-bottom-left-radius: 12px;
  border-bottom-right-radius: 12px;
  & > li {
    height: 50px;
    display: inline-flex;
    align-items: center;
    border-radius: 15px;
    padding: 0 20px;
    text-decoration: none;
    > a {
      text-decoration: none;
      color: black;
      font-weight: bold;
    }
  }
}
