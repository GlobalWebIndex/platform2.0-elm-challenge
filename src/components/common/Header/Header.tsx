import { Routes } from 'constants/routes';
import useActiveHeader, { HeaderTabEnum } from 'hooks/useActiveHeader';
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.module.scss';
import HeaderTab from './HeaderTab/HeaderTab';

const Header = () => {
  const { activeTab } = useActiveHeader();

  return (
    <nav className={styles.content}>
      <ul>
        <HeaderTab isActive={activeTab === HeaderTabEnum.Home} to={Routes.home.index} text="Home" />
        <HeaderTab isActive={activeTab === HeaderTabEnum.Breeds} to={Routes.breeds.index} text="Breeds" />
        <HeaderTab isActive={activeTab === HeaderTabEnum.Favourites} to={Routes.favourites.index} text="Favourites" />
      </ul>
    </nav>
  );
};

export default Header;
