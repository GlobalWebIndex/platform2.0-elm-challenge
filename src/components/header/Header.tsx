import { Flex, Stack } from '@chakra-ui/react'
import { FC } from 'react'
import MenuLinks from './MenuLinks'

const Header: FC = () => (
  <Flex
    as='nav'
    align='center'
    justify='space-between'
    w='100%'
    mb={8}
    p={8}
    bg={['primary.500']}
    color={['primary.700']}
  >
    <Stack spacing={8} align='center' direction={['row']}>
      <div>LOGO</div>
      <MenuLinks />
    </Stack>
  </Flex>
)

export default Header
