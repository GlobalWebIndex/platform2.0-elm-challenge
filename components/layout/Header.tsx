import { LanguageSwitcher } from 'components/layout/header/LanguageSwitcher';
import { Menu } from 'components/layout/header/Menu';
import { ThemeSwitcher } from 'components/layout/header/ThemeSwitcher';
import { menuData } from "configuration/dummy";
import { useState } from 'react';

export const Header = () => {
    const [showLanguages, setShowLanguages] = useState(false);

    return (
        <>
            <header className="fixed h-[60px] md:h-[80px] items-center w-full flex py-4 px-4 md:px-8 justify-between z-20 backdrop-blur-lg bg-white/80 dark:bg-black/80">
                <LanguageSwitcher {...{ showLanguages }} {...{ setShowLanguages }} />
                <Menu items={menuData.items} {...{ showLanguages }} />
                <ThemeSwitcher />
                <div className="absolute bottom-0 left-0">
                    <div id="nprogress-placeholder" className="bg-gray-200 dark:bg-gray-800 bottom-0 w-screen h-[1px] overflow-hidden opacity-30" />
                </div>
            </header>
        </>
    )
};