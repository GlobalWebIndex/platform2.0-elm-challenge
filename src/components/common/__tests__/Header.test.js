import { render, screen } from "@testing-library/react";
import Header from "../Header";

describe("<Header/>", () => {
  it("renders", () => {
    const { container } = render(<Header title="A title" />);

    expect(container).toMatchSnapshot();
  });

  it("does not render without title", () => {
    render(<Header />);

    expect(screen.queryByTestId("header")).not.toBeInTheDocument();
  });
});
