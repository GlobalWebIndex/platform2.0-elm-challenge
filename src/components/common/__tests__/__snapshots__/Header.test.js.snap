// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders 1`] = `
<div>
  <header
    class="title"
    data-testid="header"
  >
    <h1>
      A title
    </h1>
  </header>
</div>
`;
